pkg/services/automation/generator.go

```diff
@@ -399,6 +403,19 @@ func CreateRepoSecretName(gitSourceURL gitproviders.RepoURL) GeneratedSecretName
 	return GeneratedSecretName(lengthConstrainedName)
 }
 
```

Comment:
I think it would be nice to call out in a comment here that the two source names need to be distinct (maybe with a reference to the bug) so that it won't get inadvertently undone later by someone without context.

Reply:
Added the comment.